pgmpy/factors/CPD.py

```diff
@@ -117,6 +117,21 @@ def __init__(self, variable, variable_card, values,
             raise TypeError("Values must be a 2D list/array")
 
         super(TabularCPD, self).__init__(variables, cardinality, values.flatten('C'))
+        
+    def __repr__(self):
+        if not self.evidence:
+            str_1 = ""
+            str_1 += '<TabularCPD representing P(' + str(self.variable) + ':' + str(self.variable_card) + ')>'
+            return str_1
+
+        else:
+            str_1 = ""
+            str_1 += '<TabularCPD representing P(' + str(self.variable) + ':' + str(self.variable_card) + ' |'
+            str_2 = ""
+            for i in range(len(self.evidence)):
```

Comment:
Don't add to ```str_2``` in each iteration. Adding to any immutable object creates a new immutable object. Instead try to use string formatting as much as possible. Furthermore, the ```for``` loop can also be replaced by list comprehension as

```python
str_2 = ' '.join(['{variable}:{cardinality}'.format(variable=variable, cardinality=cardinality)
                  for variable, cardinality in zip(self.evidence, self.evidence_card)])
```